onnxruntime/python/tools/transformers/models/phi2/convert_to_onnx.py

```diff
@@ -0,0 +1,571 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import argparse
+import logging
+import os
+from enum import Enum
+
+import onnx
+import torch
+from benchmark_helper import Precision
+from dynamo_onnx_helper import DynamoOnnxHelper
+from onnx import ModelProto, TensorProto, helper
+from transformers import AutoConfig, AutoModelForCausalLM
+
+from onnxruntime.quantization.matmul_4bits_quantizer import MatMul4BitsQuantizer
+
+
+class AttentionOpType(Enum):
+    Attention = "Attention"
+    MultiHeadAttention = "MultiHeadAttention"
+    GroupQueryAttention = "GroupQueryAttention"
+
+    def __str__(self):
+        return self.value
+
+
+def env_reset():
+    for flag in ["AttentionOpType"]:
+        if flag in os.environ:
+            del os.environ[flag]
+
+
+class ConvertPhi2ToONNX(DynamoOnnxHelper):
+    def __init__(
+        self,
+        device: torch.device,
+        model_class: str = "microsoft/phi-2",
+        cache_dir: str = "./cache",
+    ):
+        self.model_class = model_class
+        self.device = device
+        self.cache_dir = cache_dir
+        self.phi_config = AutoConfig.from_pretrained(self.model_class, trust_remote_code=True, cache_dir=self.cache_dir)
+        self.phi_model = None
+        self.batch_size = 2
+        self.sequence_length = 8
+        self.phi2_edge_dict = self.get_phi2_edge_dict(self.phi_config)
+        self.attn_op_type = None
+        self.precision = None
+        self.block_size = 16
+        self.accuracy_level = None
+
+    def set_quantization_params(self, block_size: int, accuracy_level: int):
```

Comment:
nit: accuracy_level can also be None. a type annotation like this would be more accurate:
```suggestion
# at the beginning of the file
from __future__ import annotations
# ...
    def set_quantization_params(self, block_size: int, accuracy_level: int | None):
```